### PR TITLE
Edit challenge

### PR DESCRIPTION
**Pull request by GouadriaHanen**

### CODE REVIEW

File: challenges-webapp/src/main/resources/locale/portlet/Challenges_en.properties [hunk 1]
```diff
@@ -43,9 +44,12 @@ challenges.status.starts=Starts
...
 challenges.edit=Edit challenge
-challenges.challengeCreateSuccess= Challenge Created Successfully !
+challenges.challengeUpdateSuccess= Challenge updated successfully !
+challenges.challengeCreateSuccess= Challenge created successfully !
 challenges.challengeDateError=Star date must be less then end Date!
 challenges.details=Challenge details
 challenges.label.description=Description
 challenges.label.audience=Audience space
 challenges.label.managers=Challenge Manager(s)
+
```
Comment (Rdidmohamedelhadi) [inline challenges-webapp/src/main/resources/locale/portlet/Challenges_en.properties]: delete this please !!

File: challenges-webapp/src/main/webapp/vue-app/challenges/components/ChallengeAssignment.vue [hunk 2]
```diff
@@ -84,7 +86,7 @@ export default {
       menu: false,
       space: {},
       menuId: `AssigneeMenu${parseInt(Math.random() * 10000).toString()}`,
```
Comment (Rdidmohamedelhadi) [inline challenges-webapp/src/main/webapp/vue-app/challenges/components/ChallengeAssignment.vue]:
```suggestion
      menuId: `AssigneeMenu${parseInt(Math.random() * 10000)}`,
```

File: challenges-webapp/src/main/webapp/vue-app/challenges/components/ChallengesList.vue [hunk 3]
```diff
@@ -7,10 +7,9 @@
           :key="challenge.id"
           class="mb-4"
           cols="12"
-          md="6"
-          lg="4"
-          xl="3">
-          <challenge-card :challenge="challenge" />
+          sm="3"
+          md="3">
+          <challenge-card :challenge="challenge" @edit="emitEditChallenge($event)" />
```
Comment (Rdidmohamedelhadi) [inline challenges-webapp/src/main/webapp/vue-app/challenges/components/ChallengesList.vue]:
```suggestion
          <challenge-card :challenge="challenge" @edit="this.$emit('edit-challenge',challenge);" />
```

File: challenges-webapp/src/main/webapp/vue-app/challenges/components/ChallengesList.vue [hunk 4]
```diff
@@ -27,6 +26,11 @@ export default {
         return [];
       },
     }
+  },
+  methods: {
+    emitEditChallenge(challenge){
+      this.$emit('edit-challenge',challenge);
+    }
   }
```
Comment (Rdidmohamedelhadi) [inline challenges-webapp/src/main/webapp/vue-app/challenges/components/ChallengesList.vue]:
```suggestion
  }
  }
```

File: challenges-webapp/src/main/webapp/vue-app/challenges/components/ChallengeDrawer.vue [hunk 5]
```diff
@@ -221,6 +327,23 @@ export default {
...
+        this.challenge.managers = this.getManagersIds(this.challenge.managers);
+      }
+      this.$challengesServices.updateChallenge(this.challenge).then(() =>{
+        this.$root.$emit('show-alert', {type: 'success',message: this.$t('challenges.challengeUpdateSuccess')});
+        this.close();
+        this.challenge = {};
+      })
+        .catch(e => {
+          this.$root.$emit('show-alert', {type: 'error',message: String(e)});
+        });
```
Comment (Rdidmohamedelhadi) [inline challenges-webapp/src/main/webapp/vue-app/challenges/components/ChallengeDrawer.vue]: change this  in saveChallenge function please

File: challenges-webapp/src/main/webapp/vue-app/challenges/components/ChallengeDatePicker.vue [hunk 6]
```diff
@@ -74,6 +88,14 @@ export default {
...
       this.$emit('endDateChanged',new Date(date));
     },
+    getFormatDate(date) {
+      const format = {
+        year: 'numeric',
+        month: 'long',
+        day: 'numeric'
+      };
+      return this.$dateUtil.formatDateObjectToDisplay(date, format, eXo.env.portal.language);
+    }
```
Comment (Rdidmohamedelhadi) [inline challenges-webapp/src/main/webapp/vue-app/challenges/components/ChallengeDatePicker.vue]:
```suggestion
    getFormatDate(date) {
      const lang = eXo.env.portal.language;
      const options = { month: 'long' };
      const day = String(date.getDate());
      const year = String(date.getFullYear());
      return `${date.toLocaleDateString(lang || 'en', options)} ${day}, ${year}` ;
    }
```

**Review by Rdidmohamedelhadi — DISMISSED** [review]

File: challenges-webapp/src/main/webapp/vue-app/challenges/challengesUtils.js [hunk 7]
```diff
@@ -59,4 +59,12 @@ export function getUserTimezone() {
...
   const timezoneSign = timeZoneOffset >= 0 ? '+' : '-';
   return `${timezoneSign}${timezoneHours}:${timezoneMinutes}`;
+}
+
+export function getFromDate(date) {
+  const lang = eXo.env.portal.language;
+  const options = { month: 'long' };
+  const day = String(date.getDate());
+  const year = String(date.getFullYear());
+  return `${date.toLocaleDateString(lang || 'en', options)} ${day}, ${year}` ;
```
Comment (ahamdi) [inline challenges-webapp/src/main/webapp/vue-app/challenges/challengesUtils.js:69]: Please consider languages with country variant like pt_BR or es_ES

**Review by ahamdi — APPROVED** [review]

**Pull request merged**